C/Makefile

```diff
@@ -5,8 +5,16 @@ ifneq ($(doCheck), 1)
 CPPFLAGS := $(CPPFLAGS) -DNDEBUG
 endif
 
+ifneq ($(strip $(SINGLE_THREADED)),)
```

Comment:
I don't understand this construction. How can I test it?

Reply:
Something like `make test SINGLE_THREADED=1` or assign `SINGLE_THREADED` to any non-empty value.

Reply:
Can you write `""` for the empty string or add a comment explaining that you are checking that `$SINGLE_THREADED` is nonempty?

Reply:
When setting `SINGLE_THREADED=1` I get
```
primitive.c:(.text+0x1ff0): undefined reference to `call_once'
```
which I think counts as a success :P

Reply:
Is this a `make clean` error?  Do you need me to add a `make clean` rule?

Reply:
I've added some comments.